elastic/kibana
# [Cloud Security] filter in only billable assets for cloud security metering task

## Summary

Count only billable assets in Cloud Security metering task for serverless, based on the list created in https://github.com/elastic/security-team/issues/8970

Closes: 
- https://github.com/elastic/security-team/issues/9146 

### Checklist

Delete any items that are not applicable to this PR.


### How to test locally

1. Update TASK_INTERVAL to a smaller value to make the metering task to run more often
2. Run Kibana with `--verbose` flag to see DEBUG logs and ES and ingest some CSPM data
3. Look for `received usage records:` log line and compare its output with the output of 
```
POST /logs-cloud_security_posture.findings_latest-default/_search
{
  "size": 0,
  "query": {
    "bool": {
      "must": [
        {
          "range": {
            "@timestamp": {
              "gte": "now-24h"
            }
          }
        },
        {
          "term": {
            "rule.benchmark.posture_type": {
              "value": "cspm"
            }
          }
        },
        {
          "terms": {
            "resource.sub_type": [
              "aws-ebs",
              "aws-ec2",
              "aws-s3",
              "aws-rds",
              "azure-disk",
              "azure-document-db-database-account",
              "azure-flexible-mysql-server-db",
              "azure-flexible-postgresql-server-db",
              "azure-mysql-server-db",
              "azure-postgresql-server-db",
              "azure-sql-server",
              "azure-vm",
              "gcp-bigquery-dataset",
              "gcp-bigquery-table",
              "gcp-compute-disk",
              "gcp-compute-instance",
              "gcp-logging-log-bucket",
              "gcp-sqladmin-instance",
              "gcp-storage-bucket"
            ]
          }
        }
      ]
    }
  },
  "aggs": {
    "unique_assets": {
      "cardinality": {
        "field": "resource.id",
        "precision_threshold": 40000
      }
    },
    "min_timestamp": {
      "min": {
        "field": "@timestamp"
      }
    }
  }
}
```

The reported `unique_assets` should be the same
